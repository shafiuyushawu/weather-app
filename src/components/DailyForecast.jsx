import { Link } from 'react-router-dom';
import { FaRegArrowAltCircleRight } from 'react-icons/fa';
import PropTypes from 'prop-types';
import { useDispatch } from 'react-redux';
import { ForecastDetails } from '../redux/weather/weatherSlice';

const DailyForecast = ({ fcast }) => {
  const dispatch = useDispatch();

  const handleClick = (date) => {
    dispatch(ForecastDetails(date));
  };

  return (
    <li className="grid-cols-2 border-2 even:bg-[#34495e] border-blue-300 p-2 hover:border-blue-500 hover:shadow-sm hover:shadow-blue-500 shadow-sm shadow-green-200">
      <Link to="/weather-details" onClick={() => handleClick(fcast.date)}>
        <div className="flex justify-between">
          <span>
            Date:
            {fcast.date}
          </span>
          <FaRegArrowAltCircleRight className="text-2xl text-blue-300 hover:text-blue-500" />
        </div>
        <div className="flex justify-center">
          <hr className="my-1 w-12 h-2  items-center" />
        </div>
        <div className="flex justify-between">
          <span className="text-white text-start">{fcast.text}</span>
          <img src={fcast.icon} alt="daily-icon" className="text-2xl" />
        </div>
      </Link>
    </li>
  );
};

DailyForecast.propTypes = {
  fcast: PropTypes.shape({
    date: PropTypes.string.isRequired,
    icon: PropTypes.string.isRequired,
    text: PropTypes.string.isRequired,
  }).isRequired,
};
export default DailyForecast;
