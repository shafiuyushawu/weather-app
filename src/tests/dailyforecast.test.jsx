import React from 'react';
import { render } from '@testing-library/react';
import configureStore from 'redux-mock-store';
import { BrowserRouter } from 'react-router-dom';
import { Provider } from 'react-redux';
import DailyForecast from '../components/DailyForecast';
import '@testing-library/jest-dom/extend-expect';

const mockStore = configureStore([]);

test('renders DailyForecast component', () => {
  const fcast = {
    date: '2023-04-22',
    icon: 'sunny.png',
    text: 'Sunny',
  };

  const store = mockStore({
    weather: {
      forecast: [fcast],
    },
  });

  const { container } = render(
    <BrowserRouter>
      <Provider store={store}>
        <DailyForecast fcast={fcast} />
      </Provider>
    </BrowserRouter>,
  );

  expect(container.firstChild).toMatchSnapshot();
});
