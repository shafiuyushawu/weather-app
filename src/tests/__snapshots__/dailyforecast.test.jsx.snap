// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders DailyForecast component 1`] = `
<li
  class="grid-cols-2 border-2 border-blue-300 p-2 hover:border-blue-500 hover:shadow-sm hover:shadow-blue-500 shadow-sm shadow-green-200"
>
  <a
    href="/weather-details"
  >
    <div
      class="flex justify-between"
    >
      <span>
        Date:
        2023-04-22
      </span>
      <svg
        class="text-2xl text-blue-300 hover:text-blue-500"
        fill="currentColor"
        height="1em"
        stroke="currentColor"
        stroke-width="0"
        viewBox="0 0 512 512"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M504 256C504 119 393 8 256 8S8 119 8 256s111 248 248 248 248-111 248-248zm-448 0c0-110.5 89.5-200 200-200s200 89.5 200 200-89.5 200-200 200S56 366.5 56 256zm72 20v-40c0-6.6 5.4-12 12-12h116v-67c0-10.7 12.9-16 20.5-8.5l99 99c4.7 4.7 4.7 12.3 0 17l-99 99c-7.6 7.6-20.5 2.2-20.5-8.5v-67H140c-6.6 0-12-5.4-12-12z"
        />
      </svg>
    </div>
    <div
      class="flex justify-center"
    >
      <hr
        class="my-1 w-12 h-2  items-center"
      />
    </div>
    <div
      class="flex justify-between"
    >
      <span
        class="text-white text-start"
      >
        Sunny
      </span>
      <img
        alt="daily-icon"
        class="text-2xl"
        src="sunny.png"
      />
    </div>
  </a>
</li>
`;
